MSAL/src/public/MSALInteractiveTokenParameters.h

```diff
@@ -62,33 +63,63 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, nullable) NSArray<NSString *> *extraScopesToConsent;
 
 /*!
- Initialize a MSALInteractiveTokenParameters with scopes.
- 
- @param scopes      Permissions you want included in the access token received
-                    in the result in the completionBlock. Not all scopes are
-                    gauranteed to be included in the access token returned.
+ A copy of the configuration which was provided in the initializer.
  */
-- (instancetype)initWithScopes:(NSArray<NSString *> *)scopes NS_DESIGNATED_INITIALIZER;
+@property (nonatomic, readonly, copy) MSALWebviewParameters *webviewParameters;
 
 #if TARGET_OS_IPHONE
 /*!
  Modal presentation style for displaying authentication web content.
  */
-@property (nullable, weak, nonatomic) UIViewController *parentViewController;
-@property (nonatomic) UIModalPresentationStyle presentationStyle;
+@property (nullable, weak, nonatomic) UIViewController *parentViewController DEPRECATED_MSG_ATTRIBUTE("Create MSALWebviewParameters and provide it to -initWithScopes:webviewParameters: instead");
+
+@property (nonatomic) UIModalPresentationStyle presentationStyle DEPRECATED_MSG_ATTRIBUTE("Create MSALWebviewParameters and provide it to -initWithScopes:webviewParameters: instead");
+
 #endif
 
 /*!
  A specific webView type for the interactive authentication flow.
  By default, it will be set to MSALGlobalConfig.defaultWebviewType.
  */
-@property (nonatomic) MSALWebviewType webviewType;
+@property (nonatomic) MSALWebviewType webviewType DEPRECATED_MSG_ATTRIBUTE("Create MSALWebviewParameters and provide it to -initWithScopes:webviewParameters: instead");
+
 /*!
  For a webviewType MSALWebviewTypeWKWebView, custom WKWebView can be passed on.
  Web content will be rendered onto this view.
  Observe strings declared in MSALPublicClientStatusNotifications to know when to dismiss.
  */
-@property (nonatomic, nullable) WKWebView *customWebview;
+@property (nonatomic, nullable) WKWebView *customWebview DEPRECATED_MSG_ATTRIBUTE("Create MSALWebviewParameters and provide it to -initWithScopes:webviewParameters: instead");
+
+#if TARGET_OS_IPHONE
```

Comment:
should this be opposite? if not TARGET_OS_IPHONE?

Reply:
Nope, it is correct: we deprecate api for iphone and in "else" we declare similar but not deprecated api for mac.

Reply:
ok :)